src/main/scala/com/cognite/spark/datasource/CdpConnector.scala

```diff
@@ -32,16 +32,22 @@ trait CdpConnector {
 
   def getProject(auth: Auth, maxRetries: Int, baseUrl: String): String = {
     val loginStatusUrl = uri"$baseUrl/login/status"
-    val jsonResult = getJson[Data[Login]](auth, loginStatusUrl, maxRetries)
+    val jsonResult = getJson[Data[Login]](auth, loginStatusUrl, maxRetries, None)
     jsonResult.unsafeRunSync().data.project
   }
 
   def baseUrl(project: String, version: String, baseUrl: String): Uri =
     uri"$baseUrl/api/$version/projects/$project"
 
-  def getJson[A: Decoder](auth: Auth, url: Uri, maxRetries: Int): IO[A] = {
+  def getJson[A: Decoder](
```

Comment:
I think it's time to refactor this to take a `config` argument, since most arguments are coming from there already, and then we wouldn't need to add `applicationId` as yet another argument.

Reply:
Then we could also make it a non-optional field and set it only once (when `config` is initialized, possibly using a default value for `applicationId`) instead of having to do `applicationId.getOrElse` in several places.

Unless I'm missing something which is quite possible. :)